Mathlib/Tactic/Rewrites.lean

```diff
@@ -231,12 +237,13 @@ elab_rules : tactic |
       if results.isEmpty then
```

Comment:
Github doesn't let me comment or suggest on the appropriate line, but: should we check that (reducible) `rfl` doesn't close the goal by itself before looking at rewrite? (Note that even `rw ... at h` runs `rfl` on the goal and can close it, so checking it is indeed applicable in every case.) For example:
```
example : 0 = 0 by rw? -- Try this: rw [← @Nat.le_zero] -- "no goals"
```
(This is technically orthogonal to this PR, and maybe should be in a different one, but was born out of thinking about `computeRfl`.)

Reply:
Hmm... I'm not sure here. I agree the behaviour of `rw?` is confusing when you present it with a goal solvable by `rfl`, but I don't think this makes it `rw?`'s job to try `rfl`.

How about we don't modify this behaviour for now, but I'm happy to consider it separately.